### PR TITLE
Support both OpenSwan and StrongSwan VPN

### PR DESCRIPTION
**Pull request by remibergsma**
This PR switches to StrongSwan when the systemvm version is newer than `17.3.12` (to be decided what version exactly). It provides a seamless upgrade from OpenSwan to StrongSwan (just Restart+Cleanup VPC) and prevents having to do a massive replace of all running routers in one go.

The UI supports both options for OpenSwan and StrongSwan.

The current build tests for OpenSwan which should still be green. We also provided test results for a run with a StrongSwan template in the comments below.

First merge the systemvm change: https://github.com/MissionCriticalCloud/systemvm-packer/pull/25 and then make sure all build slaves have this template (all other builds will fail the vpn tests). Then merge this PR and have master turn green. Finally all other open PRs should rebase on master to be able to turn green again.

Based on ACS PR 1741.

### CODE REVIEW

File: cosmic-core/systemvm/patches/debian/config/opt/cloud/bin/configure.py [hunk 1]
```diff
@@ -451,19 +451,19 @@ def process(self):
 
     def deletevpn(self, ip):
         logging.info("Removing VPN configuration for %s", ip)
-        CsHelper.execute("ipsec auto --down vpn-%s" % ip)
-        CsHelper.execute("ipsec auto --delete vpn-%s" % ip)
+        CsHelper.execute("ipsec down vpn-%s" % ip)
+        CsHelper.execute("ipsec down vpn-%s" % ip)
```
Comment (borisroman) [inline cosmic-core/systemvm/patches/debian/config/opt/cloud/bin/configure.py]: Redundant line?

File: cosmic-core/systemvm/patches/debian/config/opt/cloud/bin/configure.py [hunk 2]
```diff
@@ -451,19 +451,19 @@ def process(self):
...
         os.remove(vpnsecretsfile)
-        CsHelper.execute("ipsec auto --rereadall")
+        CsHelper.execute("ipsec reload")
 
     def configure_iptables(self, dev, obj):
         self.fw.append(["", "front", "-A INPUT -i %s -p udp -m udp --dport 500 -s %s -d %s -j ACCEPT" % (dev, obj['peer_gateway_ip'], obj['local_public_ip'])])
         self.fw.append(["", "front", "-A INPUT -i %s -p udp -m udp --dport 4500 -s %s -d %s -j ACCEPT" % (dev, obj['peer_gateway_ip'], obj['local_public_ip'])])
         self.fw.append(["", "front", "-A INPUT -i %s -p esp -s %s -d %s -j ACCEPT" % (dev, obj['peer_gateway_ip'], obj['local_public_ip'])])
-        self.fw.append(["nat", "front", "-A POSTROUTING -o %s -m mark --mark 0x525 -j ACCEPT" % dev])
+        self.fw.append(["nat", "front", "-A POSTROUTING -t nat -o %s -m mark --mark 0x525 -j ACCEPT" % dev])
```
Comment (borisroman) [inline cosmic-core/systemvm/patches/debian/config/opt/cloud/bin/configure.py]: Table nat is already defined as the first argument, right?

**Review by borisroman — COMMENTED** [review]
Please see comments

File: cosmic-core/systemvm/patches/debian/config/opt/cloud/bin/configure.py (hunk 1 above)
Comment (remibergsma) [reply cosmic-core/systemvm/patches/debian/config/opt/cloud/bin/configure.py]: Yes, and it seems this is as designed (according to ACS PR 872). Feels a bit lame, we could test it without once we see this works?

screen shot 2017-03-09 at 21 21 13 pm

File: cosmic-core/systemvm/patches/debian/config/opt/cloud/bin/configure.py (hunk 2 above)
Comment (remibergsma) [reply cosmic-core/systemvm/patches/debian/config/opt/cloud/bin/configure.py]: Indeed, this change is not needed. Will revert it, thanks.

**Comment by remibergsma** [comment]
go build

**Comment by remibergsma** [comment]
go build

**Comment by remibergsma** [comment]
Test results with StrongSwan enabled template (on NSX):
```
rbergsma@mccdpod01-hv01:/data/git/MCCT-SHARED-1/cosmic$ cat cosmic-core/test/integration/results.txt 
Test Remote Access VPN in VPC ... === TestName: test_01_vpc_remote_access_vpn | Status : SUCCESS ===ok
Test Site 2 Site VPN Across VPCs ... === TestName: test_02_vpc_site2site_vpn | Status : SUCCESS ===ok
Test Site 2 Site VPN Across redundant VPCs ... === TestName: test_03_redundant_vpc_site2site_vpn | Status : SUCCESS ===ok
Test Site 2 Site multiple VPNs Across VPCs ... === TestName: test_04_vpc_site2site_multiple_vpn | Status : SUCCESS ===ok

----------------------------------------------------------------------
Ran 4 tests in 1425.260s

OK
```

And with plain vlans:
```
rbergsma@mccdpod01-hv01:/data/git/MCCT-SHARED-2/cosmic$  cat cosmic-core/test/integration/results.txt 
Test Remote Access VPN in VPC ... === TestName: test_01_vpc_remote_access_vpn | Status : SUCCESS ===ok
Test Site 2 Site VPN Across VPCs ... === TestName: test_02_vpc_site2site_vpn | Status : SUCCESS ===ok
Test Site 2 Site VPN Across redundant VPCs ... === TestName: test_03_redundant_vpc_site2site_vpn | Status : SUCCESS ===ok
Test Site 2 Site multiple VPNs Across VPCs ... === TestName: test_04_vpc_site2site_multiple_vpn | Status : SUCCESS ===ok

----------------------------------------------------------------------
Ran 4 tests in 1840.369s

OK
```

**Comment by remibergsma** [comment]
go build

**Pull request merged**